Stop injecting anti-fingerprint protections into xml and json documents

Charlie-belmer opened the pull request:
This adds some content checks into the anti-fingerprinting script injection, and will not inject scripts if the content type is json or xml. 

Fixes issue #503

dharb reviewed (commented):
This looks good, nice fix! I just added the same fix for our GPC injection to this same PR.

kdzwinel commented on shared/data/fingerprint-protection.js:
```diff
@@ -283,6 +283,17 @@
+                }
+            } catch (e) {
+                // if we can't find content type, go ahead with injection.
+            }
```
instead of doing this check here, I think it'd be better to do it couple of lines below and avoid creating a window resize listener and the injection script in the first place e.g.

```js
if (!isXMLorJSON()) {
    window.addEventListener('resize', function () {
        const windowScript = setWindowDimensions()
        inject(windowScript, true)
    })

    const injectionScript = buildInjectionScript()
    inject(injectionScript)
}
```

or do it at the very beginning of the script and just `return` early like Dave did below.

Charlie-belmer replied:
Good thinking, made the changes.

kdzwinel reviewed (commented):
I left one comment. Also, please test if we are not breaking the PDF reader on any of the browsers 🙏

Charlie-belmer commented:
I updated and also added a couple of other content types for RSS feeds which might have display issues. The larger thing to think about here is how much of this content type selectivity we want - there are many potential content types for structured data which the browser may treat in special ways: https://www.iana.org/assignments/media-types/media-types.xhtml

It seems this change will work for the most common use cases, but it's something we may want to think about.

dharb reviewed (approved):
👍
